lib/commands/promote.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const chalk = require('chalk');
+const isMissingData = require('./validations').isMissingData;
+const max_promotions = 3;
+
+let elapsedInterval;
+let spamSeed = generateSeed()
+
+let promoteCount = 0;
+function interrupt() {
+    return promoteCount++ >= max_promotions;
+}
+// WARNING: Not cryptographically secure. Do not use any seeds generated by this generator to actually store any value.
+function generateSeed() {
+    const validChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ9'
+    return Array.from(new Array(81), (x, i) => validChars[Math.floor(Math.random() * validChars.length)]).join('')
+}
+
+function generateTransfer() {
+    return {
+        address: spamSeed,
+        value: 0,
+        message: '',
+        tag: ''
+    }
+}
+const setupPromoteCommand = (data, iotajs, refreshAccountData, vorpal) => {
+  vorpal
+    .command('promote <hash>', 'Promotes a pending transaction.  Provide an id for additional details.')
+    .option('-n <number>', 'Max number of transactions.  Default 10.')
+    .autocomplete({
+      data: () => {
+        vorpal.ui.input(`promote ${vorpal.ui.input().split(' ')[1].toUpperCase()}`);
+        if (!data.accountData) {
+          return [];
+        }
+        return data.accountData.transfers.map(b => b[0]).map(t => t.hash.slice(0, 6));
+      }
+    })
+    .action((args, callback) => {
+      if (isMissingData(['node'])) {
+        return callback();
+      }
+
+      var hash = args.hash;
+      if (hash.length < 81) {
+        if (!data.accountData) {
+          vorpal.log(chalk.red('Please provide a full transaction hash'));
+          return callback();
+        }
+        const transactionsMatchingHash = data.accountData.transfers
+          .map(b => b[0])
+          .filter(t => t.hash.indexOf(hash) !== -1);
+        if (transactionsMatchingHash.length === 0) {
+          vorpal.log(chalk.red('That hash does not match a transaction'));
+          return callback();
+        }
+        hash = transactionsMatchingHash[0].hash;
+      }
+
+      iotajs.api.getLatestInclusion([hash], (error, states) => {
+        var confirmFound = false;
```

Comment:
I prefer the filter strategy you used on line 54 above.  Let's go with something like that here again.